webapp/src/types/types.ts

```diff
@@ -150,3 +152,16 @@ export const CallAlertStatesDefault = {
         show: false,
     },
 };
+
+export type EmojiData = {
+    name: string;
+    skin?: string;
+    unified: string;
+}
+
+export type Reaction = {
+    emoji: EmojiData;
+    timestamp: number;
+    user_id: string;
+    displayName: string;
```

Comment:
I struggle to understand why we'd need to embed this vs calculating it from the `user_id` when necessary.

Reply:
This was the bug mentioned in the comments from the contribute team: when a user leaves the call, their reactions are still in the stream, but their profile is no longer in the call => their reactions in the stream switch from "Claudio Costa" to "Somebody". I don't want to remove the reaction from the stream before it times out.
See the note here: https://github.com/mattermost/mattermost-plugin-calls/blob/b894245cef01939418db7c9d59151ddb8ee1c92c/webapp/src/websocket_handlers.ts#L207

Reply:
Surely we could still access the profile through the main redux store? It's not like they left the team or something.

Reply:
This was an interesting one. It turns out we don't have access to all the profiles in the ExpandedView -- since it's a `NeedsTeamRoute` mounted component, we would have to manually request all the known user profiles from the backend ourselves (possibly hundreds), and dispatch them to the store, so that they're there when we call `getUsers` selector. Since we don't actually honour the user's teammate displayname setting (should I make a ticket to fix that?), and we don't actually use the store's up-to-date version of the user profiles, we don't display any changes to the user display names once they've joined a call. They can change their names, or we can change our teammate displayname preferences, and there won't be any change in the Calls components. 
So, given all of that -- I like this as a performant way of calculating and storing the displayname once, rather than recalculating it on every re-render (with no change).

Reply:
Alright. At some point we'll have to request the profiles of any user in a call if we ever migrate to use the standalone view for this, although not something we need to deal with today.